### PR TITLE
feat(api): Webhook alert channel(v2)

### PR DESCRIPTION
**Pull request by dmurray-lacework**
***Issue***: https://lacework.atlassian.net/browse/ALLY-651

***Description:***
Migrate the Webhook alert channel to use the V2 API.

### CODE REVIEW

File: api/alert_channels_webhook.go [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+	Data WebhookAlertChannelV2 `json:"data"`
+}
+
+type WebhookAlertChannelV2 struct {
+	v2CommonIntegrationData
+	Data WebhookDataV2 `json:"data"`
+}
+
+type WebhookDataV2 struct {
+	Url string `json:"webhookUrl"`
```
Comment (vatasha) [inline api/alert_channels_webhook.go:50]: nit: I feel like the field name should match the json field name.

Comment (vatasha) [inline api/alert_channels_webhook.go:50]:
```suggestion
	webhookUrl string `json:"webhookUrl"`
```

File: api/alert_channels_webhook_test.go [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+	)
+	assert.Nil(t, err)
+
+	response, err := c.V2.AlertChannels.GetWebhook(intgGUID)
+	assert.Nil(t, err)
+	assert.NotNil(t, response)
+	assert.Equal(t, intgGUID, response.Data.IntgGuid)
+	assert.Equal(t, "integration_name", response.Data.Name)
+	assert.True(t, response.Data.State.Ok)
+	assert.Contains(t, response.Data.Data.Url, "https://hooks.webhook.com/?api-token=12345")
```
Comment (vatasha) [inline api/alert_channels_webhook_test.go:61]: nit: I think you can use an assert.Equals here not sure why there is a containment check.

Comment (vatasha) [inline api/alert_channels_webhook_test.go:61]:
```suggestion
	assert.Equal(t, "https://hooks.webhook.com/?api-token=12345", response.Data.Data.Url)
```

File: api/alert_channels_webhook_test.go [hunk 3]
```diff
@@ -0,0 +1,135 @@
...
+	assert.Equal(t, "Webhook", webhookAlertChan.Type, "a new Webhook alert channel should match its type")
+	assert.Equal(t, 1, webhookAlertChan.Enabled, "a new Webhook alert channel should be enabled")
+	webhookAlertChan.IntgGuid = intgGUID
+
+	response, err := c.V2.AlertChannels.UpdateWebhook(webhookAlertChan)
+	assert.Nil(t, err)
+	assert.NotNil(t, response)
+	assert.Equal(t, intgGUID, response.Data.IntgGuid)
+	assert.True(t, response.Data.State.Ok)
+	assert.Contains(t, response.Data.Data.Url, "https://hooks.webhook.com/?api-token=12345")
```
Comment (vatasha) [inline api/alert_channels_webhook_test.go:111]: same comment as above

File: api/alert_channels_webhook_test.go [hunk 4]
```diff
@@ -0,0 +1,135 @@
...
+		},
+	)
+	assert.Equal(t, "integration_name", webhookAlertChan.Name, "Webhook alert channel name mismatch")
+	assert.Equal(t, "Webhook", webhookAlertChan.Type, "a new Webhook alert channel should match its type")
+	assert.Equal(t, 1, webhookAlertChan.Enabled, "a new Webhook alert channel should be enabled")
+	webhookAlertChan.IntgGuid = intgGUID
+
+	response, err := c.V2.AlertChannels.UpdateWebhook(webhookAlertChan)
+	assert.Nil(t, err)
+	assert.NotNil(t, response)
```
Comment (vatasha) [inline api/alert_channels_webhook_test.go:108]:
```suggestion
	assert.NoError(t, err)
```

File: api/alert_channels_webhook_test.go (hunk 3 above)
Comment (vatasha) [inline api/alert_channels_webhook_test.go:111]:
```suggestion
	assert.Equals(t, "https://hooks.webhook.com/?api-token=12345", response.Data.Data.Url)
```

File: api/alert_channels_webhook_test.go [hunk 5]
```diff
@@ -0,0 +1,135 @@
...
+	assert.Equal(t, intgGUID, response.Data.IntgGuid)
+	assert.True(t, response.Data.State.Ok)
+	assert.Contains(t, response.Data.Data.Url, "https://hooks.webhook.com/?api-token=12345")
+}
```
Comment (vatasha) [inline api/alert_channels_webhook_test.go:112]: nit: Can use fmt.Sprintf here instead of string concatenation

**Review by vatasha — APPROVED** [review]
LGTM 🎉 . I have some minor nits but nothing blocking.

File: api/alert_channels_webhook_test.go (hunk 4 above)
Comment (afiune) [reply api/alert_channels_webhook_test.go:108]: I don't think this is right

Comment (vatasha) [reply api/alert_channels_webhook_test.go:108]: I think this is correct. You can refer to a more complete way of doing it like so 

` actualObj, err := SomeFunction()
  if assert.NoError(t, err) {
	   assert.Equal(t, expectedObj, actualObj)
  }`

and the docs can be found here https://pkg.go.dev/github.com/stretchr/testify/assert#NoError

Comment (afiune) [reply api/alert_channels_webhook_test.go:108]: NoError is only for errors, we can change the NotNil above for this NotError but here we are trying to check that the response is not nil, which I think is the right thing.

**Review by afiune — APPROVED** [review]
There are some suggestions that need to be addressed.

File: api/alert_channels_webhook_test.go (hunk 5 above)
Comment (dmurray-lacework) [reply api/alert_channels_webhook_test.go:112]: There's no concatenated string in this segment of code.

**Pull request merged**